src/Services/Air/Air.js

```diff
@@ -167,5 +167,77 @@ module.exports = (settings) => {
             .catch(() => results)
         );
     },
+
+    cancelTicket(options) {
+      return service.getTicket(options)
+        .catch(
+          (err) => {
+            if (!(err instanceof AirRuntimeError.TicketInfoIncomplete)) {
+              throw err;
+            }
+            return this.getPNRByTicketNumber({
+              ticketNumber: options.ticketNumber,
+            })
+              .then(pnr => this.importPNR({ pnr }))
+              .then(() => service.getTicket(options));
+          }
+        )
+        .then(
+          ticketData => service.cancelTicket({
+            pnr: ticketData.pnr,
+            ticketNumber: options.ticketNumber,
+          })
+        )
+        .catch(
+          err => Promise.reject(new AirRuntimeError.FailedToCancelTicket(options, err))
+        );
+    },
+
+    cancelPNR(options) {
+      return this.getTickets(options)
+        .then(
+          tickets => Promise.all(tickets.map(
+            (ticketData) => {
+              // Check for VOID
+              const allTicketsVoid = ticketData.tickets.every(
+                ticket => ticket.coupons.every(
+                  coupon => coupon.status === 'V'
+                )
+              );
+              if (allTicketsVoid) {
+                return Promise.resolve(true);
+              }
+              // Check for cancelTicket option
+              if (options.cancelTickets !== true) {
+                return Promise.reject(new AirRuntimeError.PNRHasOpenTickets());
+              }
+              // Check for not OPEN/VOID segments
+              const hasNotOpenSegment = ticketData.tickets.some(
+                ticket => ticket.coupons.some(
+                  coupon => 'OV'.indexOf(coupon.status) === -1
+                )
+              );
+              if (hasNotOpenSegment) {
+                return Promise.reject(new AirRuntimeError.UnableToCancelTicketStatusNotOpen());
+              }
+              return Promise.all(
+                ticketData.tickets.map(
+                  ticket => (
+                    ticket.coupons[0].status !== 'V' ? service.cancelTicket({
```

Comment:
No, this checks are to detect are all of the tickets eligible to be canceled. If we move this to cancelTicket we may run into the situation where some tickets are cancelled, and some are not.